### PR TITLE
Clarify the message "this hint will only be used by eauto"

### PR DESCRIPTION
**Pull request by Armael**
**Kind:** documentation (user message improvement)

This clarifies slightly the message "this hint will only be used be eauto", that is printed when doing `Hint Resolve` on a hint that requires instantiating evars to be applied.

The new message is: `the hint: ... will only be used by eauto, because it requires instantiating evars`.
Bikeshedding on the exact text of the clarified message is of course welcome.

### CODE REVIEW

**Comment by ppedrot** [comment]
I don't think that *instantiating* is the proper term, but I cannot see how to phrase it in a better way. Any ideas, people?

**Comment by SkySkimmer** [comment]
What's an example of when this happens?

**Comment by Armael** [comment]
Artificial example:
```coq
Hypothesis L : forall (x y z: nat),
  x = y ->
  y = z.

Hint Resolve L.
```

**Comment by aspiwack** [comment]
In a discussion on Gitter, we became convinced that speaking of instantiating was misleading here because the problem was rather evars which were _not_ instantiated. There was a bit of a back and forth to find a phrasing which was both technically correct and intuitive to grasp. We came up with the following:

> The hint Foo will only be used by eauto, because applying Foo would leave variables a, x, y as unresolved existential variables.

For instance, in your lemma `L` above:

> The hint L will only be used by eauto, because applying L would leave variable x as an unresolved existential variable.

(keeping Coq's traduction to have messages specialise to the singular form)

**Comment by SkySkimmer** [comment]
So it's because there is a variable which appears in an hypothesis yet not in the goal?

**Comment by Armael** [comment]
@SkySkimmer yes, AFAICT

I like both @gmalecha and @aspiwack suggestions. However, by looking at the code I have no clue how to retrieve the names of the "unresolved variables". Could anyone give me a hint?

**Comment by Armael** [comment]
Ok, I just pushed a new version of the patch, using @aspiwack wording. I also took the liberty of re-indenting the whole function, in order to get rid of tabulations.

**Review by ppedrot — APPROVED** [review]

**Comment by Zimmi48** [comment]
Ideally, it would be great to have an output test for this, but I don't want to make your life harder as a contributor, so not a strong request.

Cf. https://github.com/coq/coq/tree/master/test-suite#adding-a-test

**Comment by Armael** [comment]
That's fine, I'll add some output tests later today.

File: tactics/hints.ml [hunk 1]
```diff
@@ -828,38 +828,50 @@ let make_exact_entry env sigma info poly ?(name=PathAny) (c, cty, ctx) =
...
+      (Some hd,
+       { pri; poly; pat = Some pat; name;
+         db = None;
+         secvars;
+         code = with_uid (Res_pf(c,cty,ctx)); })
+    else begin
+      if not eapply then failwith "make_apply_entry";
+      if verbose then begin
+        let variables =
+          if nmiss > 1 then str "variables" else str "variable"
```
Comment (Zimmi48) [inline tactics/hints.ml]: I think there is a standard plural function defined somewhere for this.

**Comment by ppedrot** [comment]
@Armael You should rebase also, to please the CI.

**Comment by gares** [comment]
I like the current formulation

**Comment by Armael** [comment]
Rebased on top of master + used `CString.plural`

**Comment by ppedrot** [comment]
The reason the message does not appear is that you use `msg_info` rather than `msg_notice`. This looks semantically correct to me, the former being for additional information and the latter for stuff that the user requested to be printed. I don't know whether we should adapt the test-suite makefile to be more verbose or just do not add an output test for this feature.

**Comment by SkySkimmer** [comment]
Maybe we could pass -verbose or whatever the option was for that one test file (the test suite already has support for per file arguments by parsing the emacs file local stuff).

**Comment by Zimmi48** [comment]
Good idea @SkySkimmer, although I realize that this is something else that is not documented as part of https://github.com/coq/coq/blob/master/test-suite/README.md.

**Comment by Zimmi48** [comment]
The first line of the test file would need to be:

```
(* -*- mode: coq; coq-prog-args: ("-verbose") *)
```

**Comment by Zimmi48** [comment]
As reported by @Armael on Gitter the `-verbose` doesn't exist.
The one we should use here is `-lv` or `-load-verbose`, however this is supposed to take the filename as argument and replace the `-l` option that is normally used to load it.
Consequently, I don't think that this solution works either. Sorry @Armael for having proposed to add a test case and delayed this PR in vain.

**Comment by ppedrot** [comment]
OK let's merge then.

**Pull request merged**